metricflow/model/data_warehouse_model_validator.py

```diff
@@ -130,46 +161,60 @@ def gen_dimension_tasks(cls, model: UserConfiguredModel) -> List[DataWarehouseVa
         on the data source to identify which have issues.
         """
 
+        render_tools = QueryRenderingTools(model=model, system_schema=system_schema)
+
         tasks: List[DataWarehouseValidationTask] = []
-        for index, data_source in enumerate(model.data_sources):
-            # generate the subtasks
-            data_source_tasks: List[DataWarehouseValidationTask] = []
-            data_source_columns: List[str] = []
-            for dimension in data_source.dimensions:
-                dim_to_query = dimension.expr if dimension.expr is not None else dimension.name
-                data_source_columns.append(dim_to_query)
-
-                data_source_tasks.append(
+        for data_source in model.data_sources:
+            if len(data_source.dimensions) > 0:
```

Comment:
I don't think we need this. Crafting the source node is pretty cheap, and everything else will get skipped by the for loop if the dimension set is empty.

Reply:
The issue is that if there are no dimensions it'll try rendering the query with an empty node list, which raises an exception. This guard was added specifically because I ran into the exception while developing. Will add a comment to that effect.

Reply:
Yeah, I figured it out eventually. Comment might be nice, or the if not: continue idiom is less jarring for people.